[Bug] Registry endpoint access_key column varchar(255) too short for long JWT tokens (e.g. F5/NGINX Plus private registry)

**Expected behavior and actual behavior:**

When configuring a registry endpoint using a JWT token as Access ID (e.g. F5/NGINX Plus `private-registry.nginx.com`), the endpoint should save successfully and replication should work.

Instead, saving the endpoint fails with "internal server error" in the UI, and triggering replication fails with:

```
ERROR: value too long for type character varying(255) (SQLSTATE 22001)
```

Harbor core log:

```
2026-05-30T08:16:22Z [ERROR] [/lib/http/error.go:58]: {"errors":[{"code":"UNKNOWN","message":"unknown: ERROR: value too long for type character varying(255) (SQLSTATE 22001)"}]}

2026-05-30T08:14:34Z [ERROR] [/pkg/reg/adapter/native/adapter.go:127]: failed to ping registry https://private-registry.nginx.com: http status code: 401, body: <html>
```

---

**Root cause:**

The `access_key` column in the `registry` table is `varchar(255)`, which is too short for JWT tokens (typically 500–2000 characters). Note: `access_secret` was already fixed to `varchar(4096)` but `access_key` was not updated.

Verified on Harbor 2.15.0:

```sql
SELECT column_name, character_maximum_length
FROM information_schema.columns
WHERE table_name = 'registry'
AND column_name IN ('access_key', 'access_secret');
```

Result:

```
 column_name   | character_maximum_length
---------------+--------------------------
 access_key    |  255    <-- too short for JWT
 access_secret | 4096    <-- already fixed
```

---

**Steps to reproduce:**

1. Go to Administration → Registries → New Endpoint
2. Provider: `Docker Registry`
3. Endpoint URL: `https://private-registry.nginx.com`
4. Access ID: paste a JWT token from F5 MyF5 portal (typically 1000+ characters)
5. Access Secret: `none`
6. Click OK → UI shows **"internal server error"**
7. If endpoint is saved and replication is triggered manually, replication fails with the `varchar(255)` error above

---

**Workaround:**

Connect to Harbor PostgreSQL and fix the column:

```sql
docker exec -it harbor-db psql -U postgres -d registry

ALTER TABLE registry
  ALTER COLUMN access_key TYPE varchar(4096);

\q
```

Then restart harbor core:

```bash
docker compose restart core
```

After applying the workaround, the endpoint saves correctly and replication succeeds.

---

**Suggested fix:**

Update the DB migration script to alter `access_key` to `varchar(4096)`, consistent with how `access_secret` was already handled.

---

**Versions:**

- Harbor version: 2.15.0
- Docker engine version: 29.4.1 (build 055a478)
- Docker Compose version: v5.1.3
- Database: PostgreSQL 15.17

---

**Additional context:**

This issue affects any registry endpoint that uses a long string as Access ID/username — not limited to F5/NGINX Plus. Any external registry that uses JWT or other long-form tokens as credentials will hit the same problem.

--- a/make/migrations/postgresql/0190_2.16.0_schema.up.sql
+++ b/make/migrations/postgresql/0190_2.16.0_schema.up.sql
@@ -1,1 +1,10 @@
 ALTER TABLE artifact_accessory ADD COLUMN IF NOT EXISTS source varchar(50) DEFAULT 'local';
+
+/*
+Increase the length of the registry access_key column so it can store long-form credentials.
+
+This keeps access_key consistent with access_secret, which is already varchar(4096).
+
+See: https://github.com/goharbor/harbor/issues/23303
+*/
+ALTER TABLE registry ALTER COLUMN access_key TYPE varchar(4096);

